### ISSUE TITLE
type_repetition_in_bounds false positive with "fully-qualified-syntax"

### ISSUE THREAD

**Issue by mathroc** [rust-lang/rust-clippy#8056]
### Summary

When using 2 fully qualified type in a where expression "based" on different types, such as :

* `<Sc::QueryT as juniper::GraphQLValue<Sc::S>>::TypeInfo: Send + Sync`
* and `<Sc::MutationT as juniper::GraphQLValue<Sc::S>>::TypeInfo: Send + Sync`

clippy triggers a type_repetition_in_bounds error. I think that's because the "base" types (`Sc::QueryT` & `Sc::MutationT`) are entirely replaced by the `juniper::GraphQLValue<Sc::S>` type and thus both considered as `juniper::GraphQLValue<Sc::S>::TypeInfo`

### Lint Name

type_repetition_in_bounds

### Reproducer

I tried this code:

```rust
pub async fn graphql<Sc>(
    req: extract::Json<GraphQLRequest<Sc::S>>,
    schema: ExtractSchema<'_, Sc::QueryT, Sc::MutationT, Sc::SubscriptionT, Sc::S>,
    context: extract::Extension<<Sc::QueryT as juniper::GraphQLValue<Sc::S>>::Context>,
) -> impl response::IntoResponse
where
    Sc: Schema,
    Sc::QueryT: juniper::GraphQLTypeAsync<Sc::S> + Send,
    <Sc::QueryT as juniper::GraphQLValue<Sc::S>>::TypeInfo: Send + Sync,
    <Sc::QueryT as juniper::GraphQLValue<Sc::S>>::Context: Send + Sync,
    Sc::MutationT: juniper::GraphQLTypeAsync<
        Sc::S,
        Context = <Sc::QueryT as juniper::GraphQLValue<Sc::S>>::Context,
    > + Send,
    <Sc::MutationT as juniper::GraphQLValue<Sc::S>>::TypeInfo: Send + Sync,
    Sc::SubscriptionT: juniper::GraphQLSubscriptionType<
        Sc::S,
        Context = <Sc::QueryT as juniper::GraphQLValue<Sc::S>>::Context,
    > + Send,
    <Sc::SubscriptionT as juniper::GraphQLValue<Sc::S>>::TypeInfo: Send + Sync,
    Sc::S: juniper::ScalarValue + Send + Sync,
```

I saw this happen:

```
this type has already been used as a bound predicate
`-W clippy::type-repetition-in-bounds` implied by `-W clippy::pedantic`
consider combining the bounds: `<Sc::MutationT as juniper::GraphQLValue<Sc::S>>::TypeInfo: Send + Sync + Send + Sync`
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_repetition_in_bounds
```

I expected to see this happen: No issue detected

### Version

```text
rustc 1.56.1 (59eed8a2a 2021-11-01)
binary: rustc
commit-hash: 59eed8a2aac0230a8b53e89d4e99d55912ba6b35
commit-date: 2021-11-01
host: x86_64-unknown-linux-gnu
release: 1.56.1
LLVM version: 13.0.0
```
